### PR TITLE
12.10 oai pmh fixes

### PR DESCRIPTION
**Pull request by kiivihal**

### CODE REVIEW

File: web-core/app/core/harvesting/OaiPmhService.scala [hunk 1]
```diff
@@ -341,13 +347,18 @@ class OaiPmhService(queryString: Map[String, Seq[String]], requestURL: String, o
 
   private def renderRecord(record: MetadataItem, metadataPrefix: String, set: String) : Elem = {
 
-    val cachedString = record.xml.get(metadataPrefix).getOrElse(throw new RecordNotFoundException(record.itemId))
+    val cachedString: String = record.xml.get(metadataPrefix).getOrElse(throw new RecordNotFoundException(record.asPmhId))
+
+    val cleanString = if (metadataPrefix equalsIgnoreCase ("mods"))
+      cachedString.replaceFirst("xmlns:mods=\"http://www.loc.gov/mods/v3\"", "xmlns:mods=\"http://www.loc.gov/mods/v3\" xsi:schemaLocation=\"http://www.loc.gov/mods/v3 http://www.loc.gov/standards/mods/v3/mods-3-4.xsd\"")
```
Comment (manuelbernhardt) [inline web-core/app/core/harvesting/OaiPmhService.scala:353]: This is a hack, and I'm tired of cleaning this kind of thing up. Please find an appropriate solution.

Comment (geralddejong) [inline web-core/app/core/harvesting/OaiPmhService.scala:353]: At the very least, mark it clearly as a hack, with a todo which is clear about why it was done and when to remove it.  When hacks are marked they can be scheduled for removal.

File: web-core/app/models/MetadataCache.scala [hunk 2]
```diff
@@ -30,6 +30,11 @@ case class MetadataItem(modified: Date = new Date(),
     systemFields.get(field.tag).getOrElse(new BasicDBList).asInstanceOf[BasicDBList].asScala.map(_.toString).toSeq
   }
 
+  def asPmhId: String = {
```
Comment (manuelbernhardt) [inline web-core/app/models/MetadataCache.scala:33]: The MetadataCache is by design meant to be agnostic to whatever it is used for. So this method should be part of the OaiPmhService, not the record itself.

Comment (kiivihal) [inline web-core/app/models/MetadataCache.scala:33]: I need to use this in the search as well to deal with the pmhId search functionality. Therefor, this was the most appropriate place of this functionality

## 

Sjoerd Siebinga
Delving B.V., Rotterdam
http://www.delving.eu (http://www.delving.eu/)
skype: sjoerdsiebinga
mobile: +31681926731

**Pull request merged**